packages/marshal/src/marshal.js

```diff
@@ -168,15 +168,25 @@ export const makeMarshal = (
      * a canonical-json stringify of the encoded form.
      *
      * @param {Passable} val
+     * @param {string} key
      * @returns {Encoding}
      */
-    const encode = val => {
+    const encode = (val, key) => {
       if (ErrorHelper.canBeValid(val)) {
+        // FIXME: We can't protect against the validator permuting the error,
+        // since it is in our contract not to harden it.
```

Comment:
"FIXME" implies that it is fixable. Is it?
```suggestion
        // Note: We can't protect against the validator permuting the error,
        // since it is in our contract not to harden it.
```